File: 202-happy-number/202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
            s = 0
            res = [] 
            while True:
                if s in res :
                    return False
                res.append(n)
                s = 0  
                for j in str(n):
                    s += (int(j) ** 2 )
                if s == 1 :
                    return True 
                n = s 

        

File: 202-happy-number/202-happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
        	set<int> st ;
	int cur_sum = 0 ;
	while(true)
	{
		auto i = st.find(n);
		if (i != st.end())
		{
			return false ;
		}
		else
		{
			st.insert(n);
			cur_sum = 0 ;
			while(n)
			{
				int d = n % 10;
				cur_sum += (d * d);
				n/=10;
			}
			if (cur_sum == 1)
				return true;
			n = cur_sum;

		}


	}


return true;

    }
};